src/main/java/com/emc/object/s3/LargeFileUploader.java

```diff
@@ -316,8 +517,29 @@ protected void configure() {
             partSize = minPartSize;
         }
 
+        // if we are resuming an MPU, size is above MPU threshold, and we don't have an uploadId to resume,
+        // find an existing upload to resume
+        if (resumeContext != null && fullSize >= mpuThreshold) {
+            // find latest upload
+            if (resumeContext.getUploadId() == null) {
+                resumeContext.setUploadId(getLatestMultipartUploadId());
+            }
+            // list existing parts
+            if (resumeContext.getUploadId() != null && resumeContext.getPartsToSkip() == null) {
```

Comment:
One scenario needs to be considered: partsToSkip isn't checked if resumeIfInitiatedAfter is provided.